Makefile

```diff
@@ -27,32 +27,126 @@ escape_double_quote = $(subst $\",$\\",$(1))
 # Utility function to require a variable to be defined
 require_defined = $(if $(value $1),$(value $1),$(error "$1 needs to be defined"))
 
+# What implementation to build
+VARIANT ?= trivial
 # Where to put build files
 OUT ?= out
 # Where to put typescript files
 TS_OUT ?= typescript
 
 .PHONY: all clean test typescript
 
-all: \
-	$(OUT)/baseDAO.tz
+trivialDAO: VARIANT = trivial
+trivialDAO: baseDAO
+
+registryDAO: VARIANT = registry
+registryDAO: baseDAO
+
+treasuryDAO: VARIANT = treasury
+treasuryDAO: baseDAO
+
+testDAO: VARIANT = test
+testDAO: baseDAO
```

Comment:
Are these targets used? AFAIU these do he same thing as their `$(OUT)/xxx.tz` equivalents, right?

Are they meant to be more convenient?

Reply:
Yes, it seems that it was left for convinence. 

Basically I initially thought that we can eliminate indvidual build section for each variant, by setting the `VARIANT` env variable. 

But it didn't work because we also needed file targets so that the dependency lookup will work for `make`.  I think it is not very useful to leave this out, so ll remove it. Let me know if you can find a reason to leave them in.

Reply:
None comes to mind, feel free to remove these

Reply:
Removed